go/vt/mysqlctl/backup.go

```diff
@@ -525,3 +526,42 @@ func scanLinesToLogger(prefix string, reader io.Reader, logger logutil.Logger, d
 		logger.Warningf("error scanning lines from %s: %v", prefix, err)
 	}
 }
+
+func ExecuteBackupInitSQL(ctx context.Context, params *BackupParams) error {
+	if params == nil || params.InitSQL == nil || len(params.InitSQL.Queries) == 0 { // Nothing to do
+		return nil
+	}
+	if len(params.InitSQL.TabletTypes) == 0 {
+		return errors.New("backup init SQL queries provided but no tablet types on which to run them")
+	}
+	if !topoproto.IsTypeInList(params.TabletType, params.InitSQL.TabletTypes) {
+		params.Logger.Infof("Skipping backup init SQL queries %q as the backup tablet type %s is not in the provided list %s", strings.Join(params.InitSQL.Queries, ", "), params.TabletType, topoproto.MakeStringTypeCSV(params.InitSQL.TabletTypes))
+		return nil
+	}
+	initTimeout, ok, err := protoutil.DurationFromProto(params.InitSQL.Timeout)
+	if err != nil {
+		return vterrors.Wrapf(err, "invalid init SQL timeout value provided: %v", params.InitSQL.Timeout)
+	}
+	if !ok || initTimeout == 0 {
+		return vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "backup init SQL queries provided but no timeout provided -- this is dangerous and not allowed")
+	}
+	queriesCSV := strings.Join(params.InitSQL.Queries, ", ")
+	params.Logger.Infof("Executing init SQL queries %q, with a timeout of %v and fail backup on error set to %t", queriesCSV, initTimeout, params.InitSQL.FailOnError)
+	initCtx, cancel := context.WithTimeout(ctx, initTimeout)
+	defer cancel()
+	if err := params.Mysqld.ExecuteSuperQueryList(initCtx, params.InitSQL.Queries); err != nil {
+		if params.InitSQL.FailOnError {
+			return vterrors.Wrapf(err, "failed to execute init SQL queries %q and instructed to fail backup in this case", queriesCSV)
+		}
+		select {
+		case <-ctx.Done():
```

Comment:
Is this looking at the wrong context here? if `ctx` is canceled, `initCtx` will have been canceled too - but if `initCtx` is canceled (due to hitting the timeout), `ctx` won't have been cancelled necessarily and thus the `hitting the configured timeout` error message will not be logged.